Tools/yocto/rpi/fix-nativesdk-image-defs.patch

```diff
@@ -0,0 +1,406 @@
+--- a/sources/poky/meta/recipes-graphics/xwayland/xwayland_22.1.5.bb
++++ b/sources/poky/meta/recipes-graphics/xwayland/xwayland_22.1.5.bb
+@@ -23,7 +23,7 @@ OPENGL_PKGCONFIGS = "glx glamor dri3"
+ PACKAGECONFIG ??= "${XORG_CRYPTO} \
+                    ${@bb.utils.contains('DISTRO_FEATURES', 'opengl', '${OPENGL_PKGCONFIGS}', '', d)} \
+ "
+-PACKAGECONFIG[dri3] = "-Ddri3=true,-Ddri3=false"
++PACKAGECONFIG[dri3] = "-Ddri3=true,-Ddri3=false,libxshmfence"
+ PACKAGECONFIG[glx] = "-Dglx=true,-Dglx=false,virtual/libgl virtual/libx11"
+ PACKAGECONFIG[glamor] = "-Dglamor=true,-Dglamor=false,libepoxy virtual/libgbm,libegl"
+ PACKAGECONFIG[unwind] = "-Dlibunwind=true,-Dlibunwind=false,libunwind"
+diff --git a/sources/meta-webkit/conf/distro/webkitdevci.conf b/sources/meta-webkit/conf/distro/webkitdevci.conf
+new file mode 100644
+index 0000000..2b6d618
+--- /dev/null
++++ b/sources/meta-webkit/conf/distro/webkitdevci.conf
+@@ -0,0 +1,26 @@
++# Distro based on Poky
++require conf/distro/poky.conf
++
++DISTRO = "webkitdevci"
++DISTRO_NAME= "WebKit Dev@CI (Yocto Poky Based Distro)"
++
++# Expand inmediately (:=) this two variables as soon as they are assigned (now),
++# otherwise the (default) lazy expansion causes issues with BASEFILESISSUEINSTALL
++# at recipe 'base-files' due to 'metadata not deterministic' errors.
++WEBKIT_CROSS_TARGET := "${@d.getVar('BB_ORIGENV').getVar('WEBKIT_CROSS_TARGET') or '${MACHINE}'}"
++WEBKIT_CROSS_VERSION := "${@d.getVar('BB_ORIGENV').getVar('WEBKIT_CROSS_VERSION') or '${DATE}'}"
++
++DISTRO_NAME:append = " [target ${WEBKIT_CROSS_TARGET}]"
++DISTRO_VERSION:append = "_${WEBKIT_CROSS_VERSION}"
++
++DISTROOVERRIDES:append = " poky:linuxstdbase"
++
++# Distro features
++DISTRO_FEATURES:append = " opengl egl pam systemd"
++DISTRO_FEATURES:remove = "ptest"
++DISTRO_FEATURES_NATIVESDK:append = " wayland"
++
++# Use systemd
++DISTRO_FEATURES_BACKFILL_CONSIDERED = "sysvinit"
++VIRTUAL-RUNTIME_init_manager = "systemd"
++VIRTUAL-RUNTIME_dev_manager = "systemd"
+diff --git a/sources/meta-webkit/recipes-browser/images/core-image-webkit-dev-ci-tools-tests.bb b/sources/meta-webkit/recipes-browser/images/core-image-webkit-dev-ci-tools-tests.bb
+new file mode 100644
+index 0000000..dd87bf2
+--- /dev/null
++++ b/sources/meta-webkit/recipes-browser/images/core-image-webkit-dev-ci-tools-tests.bb
+@@ -0,0 +1,122 @@
++SUMMARY = "Image with all the required tools to build WPE (using the script build-webkit) \
++           and also run the tests (layout-tests, performance-tests, api-tests, etc). \
++           The SDK that gets build with 'bitbake ${IMAGENAME} -c populate_sdk' is valid \
++           for using in the host with the script build-webkit and the image itself has \
++           the tools to run the tests on the board. But the image doesn't contain \
++           webkit (wpe) itself. The purpose of this image is for using on the \
++           continous integration systems of webkit.org for WPE tests."
```

Comment:
Note: I also sent the diffs to meta-webkit as a PR here: https://github.com/Igalia/meta-webkit/pull/436
I will update this patch with the reduced delta after that PR in meta-webkit lands. But having the review there landed shouldn't block landing this

Reply:
PR merged into meta-webkit, I'm updating this patch with the reduced diff